From 0e8dc737f901e854b81383b300e592225d776cfb Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Fri, 10 Aug 2018 16:32:26 +0800
Subject: [PATCH 185/706] enetc: fix writeback bit of TX BD

The W bit of TX buffer descriptor is used to determine
if writeback occurred after completion of this frame.
It will be checked before reading TX timestamping.
This patch is to fix the definition of this bit becasue
the value was wrong and the name was not very proper.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
(cherry picked from commit 44643ae18bc59fc7c41d5f69524ad5f175ac2d16)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc.c    | 2 +-
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc.c b/drivers/net/ethernet/freescale/enetc/enetc.c
index cf5c16ccada2..8857b9736fb0 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc.c
@@ -356,7 +356,7 @@ static void enetc_get_tx_tstamp(struct enetc_hw *hw, union enetc_tx_bd *txbd,
 {
 	u32 lo, hi;
 
-	if (txbd->flags & ENETC_TXBD_FLAGS_TSTMP) {
+	if (txbd->flags & ENETC_TXBD_FLAGS_W) {
 		lo = enetc_rd(hw, ENETC_SICTR0);
 		hi = enetc_rd(hw, ENETC_SICTR1);
 		if (lo <= txbd->ext.tstamp)
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index 3d1bed83e568..0b52489c4cc2 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -383,8 +383,8 @@ union enetc_tx_bd {
 };
 
 #define ENETC_TXBD_FLAGS_L4CS	BIT(0)
+#define ENETC_TXBD_FLAGS_W	BIT(2)
 #define ENETC_TXBD_FLAGS_CSUM	BIT(3)
-#define ENETC_TXBD_FLAGS_TSTMP	BIT(4)
 #define ENETC_TXBD_FLAGS_EX	BIT(6)
 #define ENETC_TXBD_FLAGS_F	BIT(7)
 
-- 
2.17.1

